src/ui/Frame.cpp

```diff
@@ -205,25 +238,25 @@ void Frame::draw(SDL_Rect _size, SDL_Rect _actualSize) {
 	// slider filler (at the corner between sliders)
 	if (actualSize.w > size.w && actualSize.h > size.h) {
 		SDL_Rect barRect;
-		barRect.x = (_size.x + _size.w) * (float)xres / (float)Frame::virtualScreenX;
-		barRect.y = (_size.y + _size.h) * (float)yres / (float)Frame::virtualScreenY;
-		barRect.w = (sliderSize) * (float)xres / (float)Frame::virtualScreenX;
-		barRect.h = (sliderSize) * (float)yres / (float)Frame::virtualScreenY;
+		barRect.x = scaledSize.x + scaledSize.w;
+		barRect.y = scaledSize.y + scaledSize.h;
+		barRect.w = sliderSize * (float)xres / (float)Frame::virtualScreenX;
+		barRect.h = sliderSize * (float)yres / (float)Frame::virtualScreenY;
 		// TODO different border styles
 		if (border > 0) {
 			switch (borderStyle) {
 			case BORDER_FLAT:
-				drawRect(&barRect, color, (Uint8)(color>>mainsurface->format->Ashift));
+				drawDepressed(barRect.x, barRect.y, barRect.x + barRect.w, barRect.y + barRect.h);
 				break;
 			case BORDER_BEVEL_HIGH:
-				drawRect(&barRect, color, (Uint8)(color>>mainsurface->format->Ashift));
+				drawDepressed(barRect.x, barRect.y, barRect.x + barRect.w, barRect.y + barRect.h);
 				break;
 			case BORDER_BEVEL_LOW:
-				drawRect(&barRect, color, (Uint8)(color>>mainsurface->format->Ashift));
+				drawDepressed(barRect.x, barRect.y, barRect.x + barRect.w, barRect.y + barRect.h);
 				break;
 			}
 		} else {
-			drawRect(&barRect, color, (Uint8)(color>>mainsurface->format->Ashift));
+			drawDepressed(barRect.x, barRect.y, barRect.x + barRect.w, barRect.y + barRect.h);
```

Comment:
Actually, it looks to me like even the else is doing the same thing, so we could entirely get rid of `if (border > 0) { ... } else ...` and instead just do `drawDepressed(barRect.x, barRect.y, barRect.x + barRect.w, barRect.y + barRect.h);`?

Unless my eyes accidentally scanned over a difference.

Reply:
The different cases are for different border styles which are not fully implemented.